### DISCUSSION TITLE
Cannot easily copy password unless edit is enabled

### DISCUSSION THREAD

**Discussion by windweaver828** [dani-garcia/vaultwarden#4941] [Q&A] [answered]
Hi, I'm not sure where to post this, sorry, I don't normally participate in communities much.

I am referencing a setup that uses organizations and collections below.

So in the webbrowser at least, unless the user has the ability to edit passwords, the copy button is not there. Is this a bug we can fix, intended behavior, or an issue of the bitwarden extension itself? And should I post this as a bug? I am hesitant to do that because I am not sure where the fault lies, or if this is how it was meant to be.

Thank you!

**Comment by Gerardv514** [comment, marked as answer]
That is probably an issue upstream and needs to be reported to bitwarden. If it’s expected behavior then would be an enhancement request.

**Comment by stefan0xC** [comment]
According to [this permissions overview](https://bitwarden.com/help/user-types-access-control/#permissions) you should be able to copy the password as long as you have any of `Can view`, `Can edit` or (in the future) `Can manage` permission. If the permission is `Can view, except passwords` or `Can edit, except passwords` it hides the password, so that's probably the issue? If not please post more infos about your system (which version, etc).

**Comment by Gerardv514** [comment]
Actually I’m a bit confused because the issue above starts out saying in the web browser you can’t copy a password, but then state this an issue with the bitwarden extension. 

My interpretation, when referencing using web browser, is that you’re using the web page/web vault. By talking about extension that’s using a browser add on program, called bitwarden extension. 

So this issue of not being able to copy is in the extension of the web browser right?

My reference above was due to fact I’ve never recalled seeing the option of a one button copy in web vault. You had to click the 3 dots and then choose copy password. So functionality to copy a password is there, both in the web vault and bitwarden extension. The variable here is the organization and permissions in which I am not sure it that works or not.

**Reply by windweaver828** [reply]
Correct I am referring to the bitwarden extension for chrome, so sorry I didn't clarify this. I figured this was most likely something to do with the extension, but I just was not sure whether vaultwarden or the extension itself was in control of whether the copy button shows up or not. I was planning to get some pictures up today which would have made it very clear what I am referring to, but have not been able to find time. I would say close this though, I think it is a pretty safe assumption that this is the extension in control of the copy button. Maybe I will report this to bitwarden later as a suggestion/enhancement.

**Reply by Gerardv514** [reply]
More importantly, what is the permissions of the items in question. Are they can view/edit?

**Reply by windweaver828** [reply]
Can edit, except password

Only if I turn it to Can Edit does it give the copy button

You would expect the copy button to be there as long as view permissions are there though. Im not sure of any good reason why anyone would be able to view but not copy a password.

I have found though, if you highlight the dots (hidden password) and use ctrl+c it will copy it but right click and copy wont work if I remember right.

**Reply by Gerardv514** [reply]
From the bitwarden link @stefan0xC provided

Can edit, except passwords:
The user or group can add new items, remove existing items, and edit existing items in the collection, except hidden fields like passwords.
Users may still use passwords via auto-fill.
Hiding passwords prevents easy copy-and-paste, however it does not completely prevent user access to this information. Treat hidden passwords as you would any shared credential.

Seems like this part details what you’re seeing:
“Prevents easy copy past but doesn’t completely prevent user to access”

In this case this would have to be an upstream feature request since it appears to be working
As intended.

**Reply by windweaver828** [reply]
Hmm, fair enough I reckon! Odd choice but hey its how they want it I guess, working as intended. I'll close this discussion now. Thanks again for finding that!
